lib/engine/game/g_1849/step/buy_train.rb

```diff
@@ -38,7 +38,7 @@ def can_sell?(entity, bundle)
 
           def buyable_trains(entity)
             # Cannot buy E-train without E-token
-            trains_to_buy = super
+            trains_to_buy = super.dup
```

Comment:
There's no point doing this. Both `reject` and `uniq` will be making copies of the array, nothing in the original array is altered.

Reply:
ah, of course. Fixed!